src/Model/Casts/BackedEnumCast.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Mongolid\Model\Casts;
+
+use BackedEnum;
+use Mongolid\Model\Casts\Exceptions\InvalidTypeException;
+use Throwable;
+
+class BackedEnumCast implements CastInterface
+{
+    /**
+     * @param class-string<BackedEnum> $backedEnum
+     */
+    public function __construct(
+        private string $backedEnum
+    ) {
+    }
+
+    public function get(mixed $value): ?BackedEnum
```

Comment:
```suggestion
    public function get(int|string $value): ?BackedEnum
```
Here you could set the types, in the case of the from method, it only accepts int or string

Reply:
Como a classe implementa uma interface, não é possível especializar o parâmetros do método, apenas o retorno.

A validação de tipo vai acabar sendo feita pelo método from.

Reply:
Could you put a docblock?

Reply:
Feito